simulate-event.js

```diff
@@ -262,7 +271,18 @@ module.exports = function (element, type, options) {
     throw new SyntaxError('Unsupported event type');
   }
 
+  var original = options;
   var eventType = eventTypes[type];
+
+  // In IE11, the Keyboard event does not allow setting the
+  // keyCode property, even with Object.defineProperty,
+  // so we have to use a UIEvent.
+  var ua = window.navigator.userAgent;
+  var msie = ua.indexOf('MSIE ');
```

Comment:
From what I could find this is the most robust way and works for fairly old versions of IE.